### DISCUSSION TITLE
Same Message Group Tasks executing in parallel in Celery with Amazon SQS FIFO

### DISCUSSION THREAD

**Discussion by manish-patel-noovosoft** [celery/celery#8349] [Q&A]
I have a setup where I'm using Celery as the task queue with Amazon SQS FIFO. My goal is to ensure sequential processing of tasks within the same message group ID, while allowing tasks with different message group IDs to be processed in parallel. However, despite following the recommended configurations and understanding the behavior of SQS message groups, I'm experiencing parallel processing of tasks within the same message group by multiple Celery worker processes. How can I ensure that tasks with the same message group ID are processed sequentially by a single worker process, while maintaining parallel processing for tasks with different message group IDs?

Some extra details (for reference) : For celery I haven't used --concurrency setting, so it is by default spawning 4 pool processes (no. of cores). I am passing message group id using following syntax :

```
    message_properties = {
        "MessageGroupId": f"{supplier_id}"
    }
    celery_task.s(
        param1, **message_properties
    ).apply_async(**message_properties)
```

and I have made sure the queue is fifo and it ends with .fifo . additional settings - { 'polling_interval': 60, 'wait_time_seconds': 10, 'visibility_timeout': 600 }
<img width="1440" alt="image" src="https://github.com/celery/celery/assets/78404651/9a7f5bbf-297d-477b-b023-27678c361ed7">

**Comment by TITANHACKY** [comment]
Hi @manish-patel-noovosoft did you find any solution or fix for this?
